cflib/localization/lighthouse_initial_estimator.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+#
+# ,---------,       ____  _ __
+# |  ,-^-,  |      / __ )(_) /_______________ _____  ___
+# | (  O  ) |     / __  / / __/ ___/ ___/ __ `/_  / / _ \
+# | / ,--'  |    / /_/ / / /_/ /__/ /  / /_/ / / /_/  __/
+#    +------`   /_____/_/\__/\___/_/   \__,_/ /___/\___/
+#
+# Copyright (C) 2022 Bitcraze AB
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, in version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+import numpy as np
+import numpy.typing as npt
+
+from .ippe_cf import IppeCf
+from cflib.localization.lighthouse_types import LhBsCfPoses
+from cflib.localization.lighthouse_types import LhCfPoseSample
+from cflib.localization.lighthouse_types import Pose
+
+
+class LighthouseInitialEstimator:
+    """
+    Make initial estimates of base station and CF poses useing IPPE (analytical solution).
+    The estimates are not good enough to use for flight but is a starting point for further
+    Calculations.
+    """
+
+    @classmethod
+    def estimate(cls, matched_samples: list[LhCfPoseSample], sensor_positions: npt.ArrayLike) -> LhBsCfPoses:
+        """
+        Make a rough estimate of the poses of all base stations and CF poses found in the samples.
+
+        The pose of the Crazyflie in the first sample is used as a reference and will define the
+        global reference frame.
+
+        :param matched_samples: A list of samples with lihghthouse angles.
+        :param sensor_positions: An array with the sensor positions on the lighthouse deck (3D, CF ref frame)
+        :return: a
+        """
+
+        bs_poses_ref_cfs = cls._angles_to_poses(matched_samples, sensor_positions)
+
+        # Use the first CF pose as the global reference frame
+        bs_poses: dict[int, Pose] = bs_poses_ref_cfs[0]
+        # TODO Do not use first sample as reference, pass it in as a parameter
```

Comment:
Should this TODO be an issue? When will we ever get around to do this? Could it be done as part of this PR? Could the TODO be removed?

TODO in code is sort of a pet peave of mine, they usually stay forever :)